### PR TITLE
Experiments basic instrumentation

### PR DESCRIPTION
**Pull request by liyiy**
## Changes

*Please describe.*  
*If this affects the frontend, include screenshots.*  

1. experiment new button clicked
2. experiment created (when user hits "save")
3. experiment launched (when user hits "launch")
4. experiment completed (when user hits "stop experiment")

## How did you test this code?

*Please describe.*

### CODE REVIEW

**Comment by liyiy** [comment]
Feel free to update/add to this PR @neilkakkar

**Comment by neilkakkar** [comment]
Added some moaar data. @paolodamico have a look too!

Hmm, I think it's still going to be hard to answer questions like: What's the average experiment running time? We should be able to answer this using posthog, but can't, with the events selected as so^.

Another problem: Aggregation happens by user, so if an experiment was started by one user and ended by another, our target metric (conversion rate from experiment launched -> experiment concluded) is borkked.

Using groups sounds like overkill (I don't want to lose one group type to such a small part of PostHog). So, need to think of better ways to store this information. 🤔

**Comment by neilkakkar** [comment]
1. Fixed the "average running length" question by sending the duration parameter with `experiment completed`
2. Experiment launched -> Completed can be created using trends and then B/A formula. This ratio should be > 0.9

**Comment by neilkakkar** [comment]
Should be good enough for now, I think? We can add more stuff as we need it

**Comment by paolodamico** [comment]
Taking a look too

File: frontend/src/lib/utils/eventUsageLogic.ts [hunk 1]
```diff
@@ -776,5 +786,43 @@ export const eventUsageLogic = kea<
         reportRecordingPlayerSkipInactivityToggled: ({ skipInactivity }) => {
             posthog.capture('recording player skip inactivity toggled', { skip_inactivity: skipInactivity })
         },
+        reportNewExperimentButtonClicked: () => {
+            posthog.capture('new experiment button clicked')
```
Comment (paolodamico) [inline frontend/src/lib/utils/eventUsageLogic.ts:790]: Perhaps this one we want to rely on autocapture? Or Pageview on `/new` ? You could've come up from a direct link.

File: frontend/src/lib/utils/eventUsageLogic.ts [hunk 2]
```diff
@@ -776,5 +786,43 @@ export const eventUsageLogic = kea<
...
+        },
+        reportExperimentCreated: ({ experiment }) => {
+            posthog.capture('experiment created', {
+                name: experiment.name,
+                id: experiment.id,
+                filters: experiment.filters,
```
Comment (paolodamico) [inline frontend/src/lib/utils/eventUsageLogic.ts:796]: We could use the same `sanitizeFilterParams` we use for insights to reduce potential exposure of sensitive info.

File: frontend/src/lib/utils/eventUsageLogic.ts [hunk 3]
```diff
@@ -776,5 +786,43 @@ export const eventUsageLogic = kea<
...
+            })
+        },
+        reportExperimentCompleted: ({ experiment, endDate, duration }) => {
+            posthog.capture('experiment completed', {
+                name: experiment.name,
+                id: experiment.id,
+                filters: experiment.filters,
+                parameters: experiment.parameters,
+                end_date: endDate.toISOString(),
+                duration,
```
Comment (paolodamico) [inline frontend/src/lib/utils/eventUsageLogic.ts:824]: I would include information on whether the experiment was significant when terminated.

**Review by paolodamico — APPROVED** [review]

File: frontend/src/lib/utils/eventUsageLogic.ts (hunk 3 above)
Comment (neilkakkar) [reply frontend/src/lib/utils/eventUsageLogic.ts:824]: this is good, definitely

File: frontend/src/lib/utils/eventUsageLogic.ts (hunk 1 above)
Comment (neilkakkar) [reply frontend/src/lib/utils/eventUsageLogic.ts:790]: Makes sense. I want to use this to track abandonment, and pageview on `/new` works nicely.

**Comment by neilkakkar** [comment]
Any idea what's going wrong with the frontend check? 🤔

**Pull request closed without merging**